delivery_tnt_oca/models/delivery_carrier.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+
+from odoo import _, fields, models
+
+from .tnt_request import TntRequest
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    delivery_type = fields.Selection(selection_add=[("tnt_oca", "TNT")])
+    tnt_oca_ws_username = fields.Char(string="ExpressConnect Username WS")
+    tnt_oca_ws_password = fields.Char(string="ExpressConnect Password WS")
+    tnt_oca_ws_account = fields.Char(string="ExpressConnect Account WS")
+    # Misc
+    tnt_product_type = fields.Selection(
+        selection=[
+            ("D", "Document (paper/manuals/reports)"),
+            ("N", "Non-document (packages)"),
+        ],
+        default="N",
+        string="Product type",
+    )
+    tnt_product_code = fields.Char(
+        compute="_compute_tnt_product_code", string="Product code",
+    )
+    tnt_product_code_d = fields.Selection(
+        selection=[
+            ("09D", "9:00 EXPRESS"),
+            ("12D", "12:00 EXPRESS"),
+            ("15D", "GLOBAL EXPRESS"),
+        ],
+        default="09D",
+        string="Product code (Docs)",
+    )
+    tnt_product_code_n = fields.Selection(
+        selection=[("15N", "GLOBAL EXPRESS"), ("48N", "ECONOMY EXPRESS")],
+        default="15N",
+        string="Product code (Non docs)",
+    )
+    tnt_product_service = fields.Char(
+        compute="_compute_tnt_product_service", string="Product service",
+    )
+    tnt_product_service_d = fields.Selection(
+        selection=[
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Docs)",
+    )
+    tnt_product_service_n = fields.Selection(
+        selection=[
+            ("EC", "Economy Express"),
+            ("EC12", "12:00 Economy Express"),
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Non docs)",
+    )
+    tnt_payment_indicator = fields.Selection(
+        selection=[("S", "Sender pays"), ("R", "Receiver pays")],
+        default="S",
+        string="Payment indicator",
+    )
+    tnt_line_of_business = fields.Selection(
+        selection=[
+            ("1", "Domestic dk to dk transfers"),
+            ("2", "international non-domestic transfers"),
+        ],
+        default="1",
+        string="Line of business",
+    )
+    tnt_collect_time_from = fields.Char(default="10:30", string="Collect time from",)
+    tnt_collect_time_to = fields.Char(default="16:00", string="Collect time to",)
```

Comment:
Usually in Odoo when you want to represent time, you use a `Float` field in the backend and `widget="float_time"` in the frontend. Did you try that?

```suggestion
    tnt_collect_time_from = fields.Float(default=10.5, string="Collect time from")
    tnt_collect_time_to = fields.Float(default=16, string="Collect time to")
```

Of course this will mean some other code changes elsewhere.